src/develop/imageop.c

```diff
@@ -2359,6 +2359,72 @@ void add_remove_mask_indicator(dt_iop_module_t *module, gboolean add)
   }
 }
 
+gboolean _iop_tooltip_callback(GtkWidget* self, gint x, gint y, gboolean keyboard_mode,
+                               GtkTooltip* tooltip, gpointer user_data)
+{
+  dt_iop_module_t *module = (dt_iop_module_t *)user_data;
+
+  const char **des = module->description(module);
+
+  if(!des) return FALSE;
+
+  GtkWidget *ht = gtk_box_new(GTK_ORIENTATION_VERTICAL, 0);
+  GtkWidget *grid = gtk_grid_new();
+  gtk_grid_set_column_homogeneous(GTK_GRID(grid), FALSE);
+  gtk_grid_set_column_spacing(GTK_GRID(grid), DT_PIXEL_APPLY_DPI(10));
+  gtk_widget_set_hexpand(grid, FALSE);
+
+  GtkWidget *label = gtk_label_new(des[0]?des[0]:"");
+  // if there is no more description, do not add a separator
+  if(des[1]) gtk_widget_set_name(label, "section_label");
+  gtk_box_pack_start(GTK_BOX(ht), label, FALSE, FALSE, 0);
+
+#ifdef _WIN32
+  // TODO: a windows dev is needed to find 4 icons properly rendered
+  const char *icon_purpose = "•";
+  const char *icon_input   = "•";
+  const char *icon_process = "•";
+  const char *icon_output  = "•";
+#else
+  const char *icon_purpose = "⟳";
+  const char *icon_input   = "⇥";
+  const char *icon_process = "⟴";
+  const char *icon_output  = "↦";
+#endif
+
+  const char *icons[4] = {icon_purpose, icon_input, icon_process, icon_output};
+  const char *ilabs[4] = {_("purpose"), _("input"), _("process"), _("output")};
+
+  for(int k=1; k<5; k++)
+  {
+    if(des[k])
+    {
+      label = gtk_label_new(icons[k-1]);
+      gtk_widget_set_halign(label, GTK_ALIGN_START);
+      gtk_grid_attach(GTK_GRID(grid), label, 0, k, 1, 1);
+
+      label = gtk_label_new(ilabs[k-1]);
+      gtk_widget_set_halign(label, GTK_ALIGN_START);
+      gtk_grid_attach(GTK_GRID(grid), label, 1, k, 1, 1);
+
+      label = gtk_label_new(":");
+      gtk_widget_set_halign(label, GTK_ALIGN_START);
+      gtk_grid_attach(GTK_GRID(grid), label, 2, k, 1, 1);
+
+      label = gtk_label_new(des[k]);
+      gtk_widget_set_halign(label, GTK_ALIGN_START);
+      gtk_grid_attach(GTK_GRID(grid), label, 3, k, 1, 1);
+      gtk_widget_set_hexpand(gtk_grid_get_child_at(GTK_GRID(grid), 3, k), TRUE);
```

Comment:
Could you maybe show in a screenshot what difference this makes; I'm not seeing any and therefore cannot test if 
`gtk_widget_set_hexpand(label, TRUE);`
does the same thing.

Reply:
Don't know why... Now it seems not needed! I have spend quite some time to find a solution to avoid all cells to expand, really strange. Maybe I have changed something else since my last test with this.

Anyway, not needed so removed.